File: nhglib-core/src/io/github/movementspeed/nhglib/input/models/InputMode.java
package io.github.movementspeed.nhglib.input.models;

/**
 * Created by Fausto Napoli on 09/01/2017.
 */
public enum InputMode {
    ONCE,
    REPEAT;

    public static InputMode fromString(String value) {
        InputMode res = null;

        switch (value.toLowerCase()) {
            case "once":
                res = ONCE;
                break;

            case "repeat":
                res = REPEAT;
                break;
        }

        return res;
    }
}

File: nhglib-core/src/io/github/movementspeed/nhglib/input/enums/InputMode.kt
package io.github.movementspeed.nhglib.input.enums

/**
 * Created by Fausto Napoli on 09/01/2017.
 */
enum class InputMode {
    ONCE,
    REPEAT;

    companion object {

        fun fromString(value: String): InputMode? {
            var res: InputMode? = null

            when (value.toLowerCase()) {
                "once" -> res = ONCE
                "repeat" -> res = REPEAT
            }

            return res
        }
    }
}
